gpio/gpioreset.go

```diff
@@ -22,18 +18,27 @@ func pinctrlSet(pin string, state string) {
 	}
 }
 
-func InitGPIO() {
+func InitGateway(resetPin, powerPin int) {
+	rst := strconv.Itoa(resetPin)
+	pwr := strconv.Itoa(powerPin)
+	initGPIO(rst, pwr)
+	resetGPIO(rst)
```

Comment:
Any interestign thing about using pinctrl directly rather than our board implementations is that we don't let you map pins to names, not worried about it. It might be a feature request in the future since they are strings in your config. We'd have to add a board dependency anyway.

Does pinctrl behave as expected with physical pin numbers?

Reply:
So does that mean on a pi we're expecting 23 to mean gpio23 - physical pin 16 on a raspberry pi?

We can change the implementation after that, but we should consider a board dependency in the future if we need to use this module with a non-Raspberry Pi board.

Reply:
yeah thats correct. A board dependency would make sense, can add that as a future improvement.

Reply:
Okay, we'll see how much hardware we have to accommodate other than a Raspberry Pi in the future. Adding a dependency is breaking in the future, but users could pin to the version that does not require a configuration change.